extensions/mssql/src/ui/dialogBase.ts

```diff
@@ -147,6 +147,17 @@ export abstract class DialogBase<DialogResult> {
 		return this.createInputBox(ariaLabel, textChangeHandler, value, enabled, 'password', width);
 	}
 
+	protected createInputBoxWithProperties(ariaLabel: string, textChangeHandler: (newValue: string) => Promise<void>, properties: azdata.InputBoxProperties, customValidation?: () => Promise<boolean>): azdata.InputBoxComponent {
+		properties.ariaLabel = ariaLabel;
+		const inputbox: azdata.InputBoxComponent = this.modelView.modelBuilder.inputBox().withProps(properties).withValidation(customValidation).component();
```

Comment:
It would still be good to use the defaults the other one sets - e.g. type is text and width is DefaultInputWidth